src/metabase/api/search.clj

```diff
@@ -177,16 +164,14 @@
         (add-collection-criteria :collection_id search-ctx))))
 
 (s/defmethod ^:private create-search-query :metric
-  [_ {:keys [collection] :as search-ctx} :- SearchContext]
-  (when-not collection
-    (-> (make-honeysql-search-query Metric "metric" metric-columns-without-type)
-        (merge-name-and-archived-search search-ctx))))
+  [_ search-ctx :- SearchContext]
+  (-> (make-honeysql-search-query Metric "metric" metric-columns-without-type)
+      (merge-name-and-archived-search search-ctx)))
 
 (s/defmethod ^:private create-search-query :segment
-  [_ {:keys [collection] :as search-ctx} :- SearchContext]
-  (when-not collection
-    (-> (make-honeysql-search-query Segment "segment" segment-columns-without-type)
-        (merge-name-and-archived-search search-ctx))))
+  [_ search-ctx :- SearchContext]
+  (-> (make-honeysql-search-query Segment "segment" segment-columns-without-type)
+      (merge-name-and-archived-search search-ctx)))
 
 (defn- favorited->boolean [row]
   (if-let [fav-value (get row :favorited)]
```

Comment:
I mentioned this in Slack but in other places in the API we use `favorite` so we should use that instead here for consistency.

Reply:
also `model` instead of `type` since that's what we've historically used / are already using elsewhere